From e950a164b7b84cdb4c9dcf5ddba7da2964ab789a Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Fri, 27 Apr 2018 15:32:51 -0700
Subject: [PATCH] ni: zynq: Mender.io support for NI Ettus Research E31x series
 SDRs

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

Signed-off-by: Moritz Fischer <moritz.fischer@ni.com>

[Adopted to u-boot 2019.07]
Signed-off-by: Joerg Hofrichter <joerg.hofrichter@ni.com>

---
 configs/ni_e31x_sg1_defconfig |  5 +++++
 configs/ni_e31x_sg3_defconfig |  5 +++++
 include/configs/ni_e31x.h     | 12 ++++++------
 3 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/configs/ni_e31x_sg1_defconfig b/configs/ni_e31x_sg1_defconfig
index d7eab43915..36b2102422 100644
--- a/configs/ni_e31x_sg1_defconfig
+++ b/configs/ni_e31x_sg1_defconfig
@@ -80,4 +80,9 @@ CONFIG_USB_GADGET_PRODUCT_NUM=0x0300
 CONFIG_CI_UDC=y
 CONFIG_USB_GADGET_DOWNLOAD=y
 CONFIG_USB_FUNCTION_THOR=y
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x800000
+CONFIG_MMC=y
 CONFIG_ENV_IS_IN_MMC=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
diff --git a/configs/ni_e31x_sg3_defconfig b/configs/ni_e31x_sg3_defconfig
index 5b1d3df98c..84d0ea556c 100644
--- a/configs/ni_e31x_sg3_defconfig
+++ b/configs/ni_e31x_sg3_defconfig
@@ -80,4 +80,9 @@ CONFIG_USB_GADGET_PRODUCT_NUM=0x0300
 CONFIG_CI_UDC=y
 CONFIG_USB_GADGET_DOWNLOAD=y
 CONFIG_USB_FUNCTION_THOR=y
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x800000
+CONFIG_MMC=y
 CONFIG_ENV_IS_IN_MMC=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
diff --git a/include/configs/ni_e31x.h b/include/configs/ni_e31x.h
index 23104f4a16..cb324c019a 100644
--- a/include/configs/ni_e31x.h
+++ b/include/configs/ni_e31x.h
@@ -12,15 +12,13 @@
 
 #include <configs/zynq-common.h>
 
-#define CONFIG_SYS_MMC_ENV_DEV 0
 #define CONFIG_ENV_SECT_SIZE           CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET              0xE0000
 
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
 	"fit_image=boot/fitImage\0"	\
 	"mboard=e31x-sg3\0"	\
-	"bootargs=root=/dev/mmcblk0p2 rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
+	"bootargs=root=${mender_kernel_root} rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"           \
 	"fdt_high=0x20000000\0"         \
@@ -31,9 +29,11 @@
 		"echo Loading FIT to RAM via DHCP/TFTP && " \
 		"dhcp ${load_addr} ${tftproot}/${fit_image} && "\
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
-	"sdboot=echo Copying FIT from SD to RAM... && " \
-		"ext4load mmc 0:2 ${load_addr} ${fit_image} && " \
-		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
+	"sdboot=run mender_setup; " \
+		"echo Copying FIT from SD to RAM... && " \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
+		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb; " \
+		"run mender_try_to_recover\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
 		"tftpboot ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
